pkg/spoke/managedcluster/creating_controller.go

```diff
@@ -53,40 +53,80 @@ func NewManagedClusterCreatingController(
 }
 
 func (c *managedClusterCreatingController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
-	_, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
-	switch {
-	case errors.IsUnauthorized(err),
-		errors.IsForbidden(err) && strings.Contains(err.Error(), anonymous):
+	existingCluster, err := c.hubClusterClient.ClusterV1().ManagedClusters().Get(ctx, c.clusterName, metav1.GetOptions{})
+	if err != nil && (errors.IsUnauthorized(err) || errors.IsForbidden(err) && strings.Contains(err.Error(), anonymous)) {
 		klog.V(4).Infof("unable to get the managed cluster %q from hub: %v", c.clusterName, err)
 		return nil
-	case errors.IsNotFound(err):
-	case err == nil:
-		return nil
-	case err != nil:
+	}
+
+	if err != nil && !errors.IsNotFound(err) {
 		return err
 	}
 
-	managedCluster := &clusterv1.ManagedCluster{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: c.clusterName,
-		},
+	// create ManagedCluster if not found
+	if err != nil && errors.IsNotFound(err) {
```

Comment:
nit, err!=nil is not necessary

Reply:
> 
@qiujian16 
err != nil is necessary, because I am not check err == nil before this.

Reply:
i think 

```
if errors.IsNotFound(err) 
```
indicate err!=nil already

Reply:
> i think
> 
> ```
> if errors.IsNotFound(err) 
> ```
> 
> indicate err!=nil already

done